src/runtime/plugin.ts

```diff
@@ -30,8 +32,52 @@ export default defineNuxtPlugin(async (nuxtApp) => {
   }
 
   // Only fetch session if it was not yet initialized server-side
-  if (typeof data.value === 'undefined' && !nitroPrerender && !disableServerSideAuth) {
-    await getSession()
+  if (
+    typeof data.value === 'undefined'
+    && !nitroPrerender
+    && !disableServerSideAuth
+  ) {
+    const config = runtimeConfig.provider as ProviderLocal
+
+    if (config.type === 'local') {
+      handleLocalAuth(config)
+    }
+
+    if (!data.value) {
+      await getSession()
+    }
+  }
+
+  function handleLocalAuth(config: ProviderLocal): void {
+    const sessionCookie = useCookie<SessionCookie | null>(
+      'auth:sessionCookie'
+    )
+    const cookieToken = useCookie<string | null>(
+      config.token?.cookieName ?? 'auth.token'
+    )
+
+    if (sessionCookie?.value && !rawToken?.value && cookieToken?.value) {
+      restoreSessionFromCookie(sessionCookie, cookieToken)
+    }
+  }
+
+  function restoreSessionFromCookie(
+    sessionCookie: CookieRef<SessionCookie | null>,
+    cookieToken: CookieRef<string | null>
+  ): void {
+    try {
+      loading.value = true
+      const sessionData = sessionCookie.value
+      lastRefreshedAt.value = sessionData?.lastRefreshedAt
+      data.value = sessionData?.data
+      rawToken.value = cookieToken.value
+    }
+    catch (error) {
+      console.error('Failed to parse session data from cookie:', error)
+    }
+    finally {
+      loading.value = false
+    }
```

Comment:
I am struggling to understand as to why saving/restoring the session data in the cookie is needed in the first place? I read both the original issue #551 and the PRs around it, but still haven't found a valid reason for _this_ approach.

I think there should be a less hacky way of solving it

Reply:
I encountered an issue with page reloads after logging in while using SSG with this module, causing the authentication token to be lost. It’s possible that I misconfigured something, but in general, I need a solution that ensures the user stays logged in even when he triggers a page reload.

When using SSG and wanting users to stay logged in across page reloads, client-side storage is required. I recommend using cookies for this purpose, as they are domain-specific, secure (with attributes like `HttpOnly` and `Secure`), and automatically sent to the server with each request, making them ideal for authentication.

localStorage is vulnerable to XSS. sessionStorage is cleared on page reloads, making it unsuitable for long-term session persistence.

While all three options have their trade-offs, cookies are typically the most secure and practical for authentication across page reloads imho. In case of JWT I guess the best option would be having short time tokens with long time opaque refresh tokens.

Reply:
One possible solution is to just load the raw token from the cookie and not store the session data anywhere.
Call `getSession` if `data.value` is null.

This would fetch the session every time the page is refreshed.

Will this work?

Reply:
The issue with the above approach is that middleware protects pages based on `status` - if a user 
refreshes a protected page, the `data.value` is lost, causing `status.value` to be 'unauthenticated'. 
This triggers unwanted redirects.

To prevent this, we can determine `status.value` based on the existence of `rawToken.value` for local 
provider only. This remains GDPR compliant as only the authentication token is stored in cookies, 
with the session being fetched on each page refresh.

Any other suggestions?

Reply:
@Vijayabhaskar96 Thank you for a good explainer! I think this is the best take so far and I would absolutely be up to fixing unnecessary redirects (but without introducing any extra cookies - only using ones we have so far:

https://github.com/sidebase/nuxt-auth/blob/b1e4abad275c020daf8e227c1b4c9c7d1e485465/src/runtime/composables/local/useAuthState.ts#L36

https://github.com/sidebase/nuxt-auth/blob/b1e4abad275c020daf8e227c1b4c9c7d1e485465/src/runtime/composables/local/useAuthState.ts#L69